### PR TITLE
Make Bytes, BytesIn and ByteOut generic to fix raw type warning https://github.com/OpenHFT/Chronicle-Bytes/issues/358

### PR DESCRIPTION
**Pull request by JerryShea**

### CODE REVIEW

File: src/main/java/net/openhft/chronicle/bytes/BytesMethodReaderBuilder.java [hunk 1]
```diff
@@ -24,19 +24,19 @@
 
 @SuppressWarnings("rawtypes")
```
Comment (minborg) [inline src/main/java/net/openhft/chronicle/bytes/BytesMethodReaderBuilder.java]: Remove suppression?

File: src/main/java/net/openhft/chronicle/bytes/DistributedUniqueTimeProvider.java [hunk 2]
```diff
@@ -61,7 +61,7 @@ public static DistributedUniqueTimeProvider instance() {
     }
 
     @SuppressWarnings("rawtypes")
```
Comment (minborg) [inline src/main/java/net/openhft/chronicle/bytes/DistributedUniqueTimeProvider.java]: ditto

File: src/main/java/net/openhft/chronicle/bytes/MappedBytes.java [hunk 3]
```diff
@@ -31,9 +31,9 @@
 import java.io.FileNotFoundException;
 
 /**
- * Bytes to wrap memory mapped data.
+ * Bytes<?> to wrap memory mapped data.
  * <p>
- * NOTE These Bytes are single Threaded as are all Bytes.
+ * NOTE These Bytes<?> are single Threaded as are all Bytes.
  */
 @SuppressWarnings({"rawtypes", "unchecked"})
```
Comment (minborg) [inline src/main/java/net/openhft/chronicle/bytes/MappedBytes.java]: remove "rawtypes"?

**Comment by minborg** [comment]
I am very pleased to see this PR. Could we go through the code and try to remove as many suppressed warnings of type "rawtypes" as possible?

Some of the changes may lead to compatibility issues but in many of those cases, there is an underlying problem anyhow that should be flushed out.

**Review by minborg — APPROVED** [review]

File: src/main/java/net/openhft/chronicle/bytes/NativeBytes.java [hunk 4]
```diff
@@ -40,7 +40,7 @@
 /**
  * Elastic memory accessor which can wrap either a ByteBuffer or malloc'ed memory.
  * <p>
- * <p>This class can wrap <i>heap</i> ByteBuffers, called <i>Native</i>Bytes for historical reasons.
+ * <p>This class can wrap <i>heap</i> ByteBuffers, called <i>Native</i>Bytes<?> for historical reasons.
```
Comment (JerryShea) [inline src/main/java/net/openhft/chronicle/bytes/NativeBytes.java]: this is picky but it would be good if we could undo the changes in comments
